### PR TITLE
Add SnowflakeBase, useful methods for permissions to GuildTextChannel, and jsdoc in a lot of places

### PR DESCRIPTION
**Pull request by DjDeveloperr**
And some changes I don't remember.

### CODE REVIEW

File: src/structures/message.ts [hunk 1]
```diff
@@ -63,8 +62,10 @@ export class Message extends Base {
     this.guildID = data.guild_id
```
Comment (Helloyunho) [inline src/structures/message.ts:62]: What about making `*_id` to Snowflake

File: src/structures/textChannel.ts [hunk 2]
```diff
@@ -226,16 +229,23 @@ export class TextChannel extends Channel {
 
     return res
   }
+
+  /** Trigger the typing indicator. NOT recommended to be used by bots unless you really want to. */
+  async triggerTyping(): Promise<TextChannel> {
+    await this.client.rest.api.channels[this.id].typing.psot()
```
Comment (Helloyunho) [inline src/structures/textChannel.ts:235]: `.psot()` nice

File: src/structures/textChannel.ts [hunk 3]
```diff
@@ -270,17 +280,21 @@ export class GuildTextChannel extends TextChannel {
...
   ): Promise<GuildTextChannel> {
     const body: ModifyGuildTextChannelPayload = {
       name: options?.name,
       position: options?.position,
       permission_overwrites: options?.permissionOverwrites,
-      parent_id: options?.parentID
+      parent_id: options?.parentID,
+      nsfw: options?.nsfw,
+      topic: options?.topic,
+      rate_limit_per_user: options?.rateLimitPerUser
```
Comment (Helloyunho) [inline src/structures/textChannel.ts:297]: I think we should change this `rate_limit_per_user` to `slowmode` too

File: src/utils/snowflake.ts [hunk 4]
```diff
@@ -9,20 +10,20 @@ export class Snowflake {
     return BigInt.asUintN(64, BigInt(this.id))
   }
 
-  get timestamp(): string {
-    return ((this.snowflake >> 22n) + 1420070400000n).toString()
+  get timestamp(): number {
+    return Number(((this.snowflake >> 22n) + 1420070400000n).toString())
```
Comment (Helloyunho) [inline src/utils/snowflake.ts:14]: You don't need to change to string if I'm right

**Review by Helloyunho — CHANGES_REQUESTED** [review]

File: src/structures/textChannel.ts (hunk 3 above)
Comment (DjDeveloperr) [reply src/structures/textChannel.ts:297]: You mean options.rateLimitPerUser? rate_limit_per_user is a name in API payload can't change lol

File: src/structures/textChannel.ts (hunk 2 above)
Comment (DjDeveloperr) [reply src/structures/textChannel.ts:235]: Typo 😅

File: src/structures/textChannel.ts (hunk 3 above)
Comment (Helloyunho) [reply src/structures/textChannel.ts:297]: oh sorry but yeah the option one

File: src/structures/embed.ts [hunk 5]
```diff
@@ -60,63 +62,116 @@ export class Embed {
...
   setType(type: EmbedTypes): Embed {
     this.type = type
     return this
   }
 
-  setURL(url: string): Embed {
-    this.url = url
+  /** Set URL of the Embed */
+  setURL(url: string): Embed
+  setURL(url: URL): Embed
```
[the remark this answers is not in the document]
Comment (DjDeveloperr) [reply src/structures/embed.ts:85]: Overrides are nicer way to represent two ways of using same method I think.

File: src/structures/message.ts (hunk 1 above)
Comment (DjDeveloperr) [reply src/structures/message.ts:62]: I'm not sure 🤔

File: src/utils/snowflake.ts (hunk 4 above)
Comment (DjDeveloperr) [reply src/utils/snowflake.ts]: Fixed.

File: src/structures/textChannel.ts (hunk 3 above)
Comment (DjDeveloperr) [reply src/structures/textChannel.ts]: Done.

File: src/structures/textChannel.ts (hunk 2 above)
Comment (DjDeveloperr) [reply src/structures/textChannel.ts]: Fixed.

File: src/structures/embed.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (Helloyunho) [reply src/structures/embed.ts:85]: yeah but it'll be same as just one `setURL(url: string | URL)`

I mean there's no any parameter name change or return type change

[the remark this answers is not in the document]
Comment (DjDeveloperr) [reply src/structures/embed.ts:85]: Alright.

[the remark this answers is not in the document]
Comment (DjDeveloperr) [reply src/structures/embed.ts]: Done.

File: src/structures/message.ts (hunk 1 above)
Comment (Helloyunho) [reply src/structures/message.ts:62]: It's just my think and we can change later so I think I can mark this comment resolved

**Review by Helloyunho — APPROVED** [review]
nice work!

**Review by AkiaCode — APPROVED** [review]

**Pull request merged**